scripts/release/download_ERA5_TEC.py

```diff
@@ -0,0 +1,331 @@
+
+    Raises
+    ------
+    ValueError
+        If account info is not set in PyAPS config file.
+    """
+    # Convert MintPy tropo model name to data archive center name
+    # NARR model included for completeness but no key required
+    MODEL2ARCHIVE_NAME = {
+        "ERA5": "CDS",
+        "ERAI": "ECMWF",
+        "MERRA": "MERRA",
+        "NARR": "NARR",
+    }
+    SECTION_OPTS = {
+        "CDS": ["key"],
+        "ECMWF": ["email", "key"],
+        "MERRA": ["user", "password"],
+    }
+
+    # Default values in cfg file
+    default_values = [
+        "the-email-address-used-as-login@ecmwf-website.org",
+        "the-user-name-used-as-login@earthdata.nasa.gov",
+        "the-password-used-as-login@earthdata.nasa.gov",
+        "the-email-adress-used-as-login@ucar-website.org",
+        "your-uid:your-api-key",
+    ]
+
+    # account file for pyaps3 < and >= 0.3.0
+    cfg_file = parent_dir / "model.cfg"
+    rc_file = Path("~").home() / ".cdsapirc"
+
+    # for ERA5: ~/.cdsapirc
+    if tropo_model == "ERA5" and rc_file.is_file():
+        pass
+
+    # check account info for the following models
+    elif tropo_model in ["ERA5", "ERAI", "MERRA"]:
+        section = MODEL2ARCHIVE_NAME[tropo_model]
+
+        # Read model.cfg file
+        cfg_file = parent_dir / "model.cfg"
+        cfg = ConfigParser()
+        cfg.read(cfg_file)
+
+        # check all required option values
+        for opt in SECTION_OPTS[section]:
+            val = cfg.get(section, opt)
+            if not val or val in default_values:
+                msg = "PYAPS: No account info found "
+                msg += f"for {tropo_model} in {section} section in file: {cfg_file}"
+                raise ValueError(msg)
+
+    return
+
+
+def get_grib_file_names(
+    slc_files: Sequence[Path], grib_dir: Path, frame_id: int
+) -> tuple[list[Path], Bbox]:
+    """Generate the grib file names for each SLC to download.
+
+    Parameters
+    ----------
+    slc_files : Sequence[Path]
+        list of paths to the slc files
+    grib_dir : Path
+        Path to the grib directory
+    frame_id : int
+        DISP-S1 frame id
+
+    Returns
+    -------
+    tuple[list[Path], Bbox]
+        The list of weather model files to be downloaded
+        and the bounding box of the frame
+    """
+    slc_file_list = group_by_date(slc_files)
+    first_date = next(iter(slc_file_list))
+    acquisition_time = get_zero_doppler_time(slc_file_list[first_date][0])
+    hour = closest_weather_model_hour(acquisition_time)
+
+    date_list = [
+        datetime.datetime.strftime(key[0], "%Y%m%d") for key in slc_file_list.keys()
+    ]
+
+    epsg, bounds = get_frame_bbox(frame_id)
+
+    if epsg != 4326:
+        bounds = transform_bounds(CRS.from_epsg(epsg), CRS.from_epsg(4326), *bounds)
+
+    snwe = (bounds[1], bounds[3], bounds[0], bounds[2])
+    area = snwe2str(snwe)
+
+    # grib file list
+    grib_files = []
+    for d in date_list:
+        if area:
```

Comment:
Is `area` ever `False`? it looks like there's always a `bounds` returned which becomes a string.

is this trying to cover a bad `frame_id`? maybe we could just error there.

Reply:
oh, or is this trying to account for like a global grib file?

Reply:
if bounds is empty for any reason, it will account for a global grib

Reply:
but I guess there will be bounds anyway because we pass frame id